src/styles/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  margin: 0 auto;
+  display: flex;
+  flex-wrap: wrap;
+  max-width: getWidth($calendar-cell-side, $gap);
+  padding: 30px $calendar-paddings;
+  gap: $gap;
+
+  @each $day, $num in $days {
+    &--start-day--#{$day} {
+      .calendar__day:first-child {
+        margin-left: firstDayShift($num, $shift);
+      }
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length--#{$length} {
+      .calendar__day:not(:nth-child(-n + #{$length})) {
```

Comment:
Too complicated selector. You said `for all elements before #{$length} don not do display: none`, when you can said `for all elements after(!!!) #{$length} do display: none`. Think how we can make selector easier.